pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -233,6 +233,11 @@ func endpointHealthStatus(svc *model.Service, e v1.Endpoint) model.HealthStatus
 	if e.Conditions.Ready == nil || *e.Conditions.Ready {
 		return model.Healthy
 	}
+	// For headless services if PublishNotReadyAddresses is set, we consider the endpoint healthy
+	// so that it is available in DNS queries in name table.
+	if svc != nil && svc.Resolution == model.Passthrough && svc.PublishNotReadyAddresses {
+		return model.Healthy
```

Comment:
Hmm, Healthy seems odd. Its not really healthy.

Note sure if it matters

Reply:
What is the status of endpointslice, if the pod is not ready/healthy

Reply:
NonReady - I think we should introduce not ready status

Reply:
> Hmm, Healthy seems odd. Its not really healthy.

Agree. This simplifies the entire flow if you use healthy. It does not matter except for the EDS case you mentioned.

Reply:
ack